src/toil/leader.py

```diff
@@ -1107,10 +1107,11 @@ def innerLoop(jobStore, config, batchSystem, toilState, jobBatcher, serviceManag
                     # Remove the job
                     if jobGraph.remainingRetryCount > 0:
                         jobBatcher.issueJob(JobNode.fromJobGraph(jobGraph))
-                        logger.debug("Job: %s is empty, we are scheduling to clean it up", jobGraph.jobStoreID)
+                        logger.debug("Job: %s is empty, we are scheduling to clean it up", jobGraph)
                     else:
                         jobBatcher.processTotallyFailedJob(jobGraph)
-                        logger.warn("Job: %s is empty but completely failed - something is very wrong", jobGraph.jobStoreID)
+                        logger.critical("Job: %s is empty but completely failed - something is "
```

Comment:
Shouldn't we raise an exception here? CRITICAL > ERROR and ERROR to means inability to continue.

If the "very wrong" part is inaccurate, I would remove that language and leave as warning.

@benedictpaten might have input on this.